test/models/provider/null_test.rb

```diff
@@ -0,0 +1,11 @@
+require "test_helper"
+
+class Provider::NullTest < ActiveSupport::TestCase
+  include ExchangeRateProviderInterfaceTest
+  include MerchantDataProviderInterfaceTest
+  include RealEstateValuationsProviderInterfaceTest
```

Comment:
As much as the dynamically typed nature of ruby gains us, this is one area where type safety would've come in handy. These tests are a good middle ground I think.

I usually wouldn't implement them. But the project being open-source means many hands with varying degrees of familiarity will be touching this system. And I don't think providers will be added often after an initial flurry. So it's easy to forget things.

Reply:
Have you played around much with the (newish) type systems in Ruby?

Reply:
I haven't! Considered those for this. But I wouldn't recommend we adopt them for the whole codebase. Seems worse to use them for a subset of the app than to implement these tests which are a simple `include` away.